code/game/machinery/computer/camera.dm

```diff
@@ -297,18 +297,73 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/computer/security/telescreen/entertai
 /obj/machinery/computer/security/telescreen/entertainment/proc/BigClick()
 	SIGNAL_HANDLER
 
+	if(!network.len)
+		usr.balloon_alert(usr, "there is nothing on TV")
+		return
+
 	INVOKE_ASYNC(src, /atom.proc/interact, usr)
 
-/obj/machinery/computer/security/telescreen/entertainment/proc/notify(on)
+///Sets the monitor's icon to the selected state, and says an announcement 
+/obj/machinery/computer/security/telescreen/entertainment/proc/notify(on, announcement)
 	if(on && icon_state == icon_state_off)
-		say(pick(
-			"Feats of bravery live now at the thunderdome!",
-			"Two enter, one leaves! Tune in now!",
-			"Violence like you've never seen it before!",
-			"Spears! Camera! Action! LIVE NOW!"))
 		icon_state = icon_state_on
 	else
 		icon_state = icon_state_off
+	if(announcement)
+		say(announcement)
+
+/// Adds a camera network ID to the entertainment monitor, and turns off the monitor if network list is empty
+/obj/machinery/computer/security/telescreen/entertainment/proc/update_shows(is_show_active, tv_show_id, announcement)
+	if(!network)
+		return
+
+	if(is_show_active)
+		network |= tv_show_id
```

Comment:
Why `|=` instead of `+=`?

Reply:
To prevent duplicate entries in case the entertainment screen already has the value in the list through some reason (subtype, admin shenanigan).

Reply:
Okay just making sure